### PR TITLE
[AIRFLOW-1196] Make trigger_dag_id a templated field of TriggerDagRunOperator

### PR DESCRIPTION
**Pull request by ghost**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1196

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Add 'trigger_dag_id' field to the list of templated fields for TriggerDagRunOperator to allow for more flexible triggering of DAGs based on the local context, plus a unit test demonstrating use of params template object

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
`tests.core.CoreTest.test_templated_trigger_dagrun`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by vfoucault** [comment]
Hi, testing this template field is quite an easy thing and improve global code quality. Check #2289 for template testing.

**Comment by ghost** [comment]
no problem, I'll take a look and add a test

**Comment by ghost** [comment]
added a unit test `test_templated_trigger_dagrun`. Travis build seems to be failing for reasons unrelated to this PR

File: tests/core.py [hunk 1]
```diff
@@ -456,6 +456,29 @@ def trigga(context, obj):
...
+            params={'trigger': 'example_bash_operator'},
+            trigger_dag_id='{{params["trigger"]}}',
+            python_callable=trigger,
+            dag=self.dag)
+
+        t.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
+
+        get_dag.assert_called_with('example_bash_operator')
+
+
```
Comment (bolkedebruin) [inline tests/core.py]: nit: too many new lines

Comment (ghost) [reply tests/core.py]: fixed!

**Comment by ghost** [comment]
rebased on latest master- looks like tests are passing now! Let me know if there is anything that needs doing.

**Comment by ghost** [comment]
rebased against master again! @bolkedebruin @vfoucault can one of you review and merge if this change looks ok (or if not let me know why)! :)

**Comment by ghost** [comment]
rebased on current master

**Comment by ghost** [comment]
maybe I've been tagging the wrong folks- @aoen I see you committed a similar PR (https://github.com/apache/incubator-airflow/pull/2549) last week, can you review and merge (:

**Pull request closed without merging**